Add a very primitive form of session persistence

SquidDev opened the pull request:
Guarded behind a config option for now (`restoreSession`), this will save the list of open computers to `<data_dir>/session.json`, and load them when the emulator starts.

Closes #118 and fixes #33 

I'm not entirely sure about the implementation, so comments welcome.

I've done a bit of looking into adding an `emu stop` command, so you can close the entire emulator (and thus restore all computers), but need to have a bit more of a thunk about how it works.

SquidDev committed:
Add a very primitive form of session persistence

Guarded behind a config option for now (restoreSession), this will
save the list of open computers to <data_dir>/session.json, and load
them when the emulator starts.

Closes #118 and fixes #33

dmarcuse commented on src/main/java/net/clgd/ccemux/emulation/SessionState.java:
```diff
@@ -0,0 +1,80 @@
+			}
+		} catch (IOException e) {
+			log.error("Cannot save session state", e);
+		}
```
This could be simplified to `try (BufferedWriter writer = ...) { ... } catch (IOException e) { ... }`

SquidDev committed:
Remove outer try

I had always assumed it didn't apply to the constructor too, but
evidently it does.

hugeblank commented:
@SquidDev This doesn't appear to fix either issues that were closed from testing on my end. Labels still don't persist, and computer 0 still opens on execution.

EDIT: aaa config option. Didn't see that sorry
